homeworks/feliistvanattila/4_exceptions_logging/to_do.py

```diff
@@ -0,0 +1,108 @@
+import logging
+
+# Logging beállítása
+logging.basicConfig(level=logging.INFO,
+                    format='%(asctime)s - %(levelname)s - %(message)s',
+                    handlers=[
+                        # logging.FileHandler ("homeworks/feliistvanattila/4_exceptions_logging/task_manager.logger"),
+                        logging.FileHandler("task_manager.logger"),
+                        logging.StreamHandler()
+                    ])
+
+# Fájl neve, ahol a feladatokat tároljuk
+# TASK_FILE = "homeworks/feliistvanattila/4_exceptions_logging/tasks.txt"
+TASK_FILE = "tasks.txt"
+
+
+def read_tasks() -> list:
+    try:
+        with open(TASK_FILE, "r") as file:
+            tasks = file.readlines()
+            tasks = [task.strip() for task in tasks]
+        logging.info("Tasks read successfully.")
+        return tasks
+    except FileNotFoundError:
+        logging.error("The task file was not found.")
+        return []
+
+
+def add_task(task: str):
+    try:
+        with open(TASK_FILE, "a") as file:
+            file.write(task + "\n")
+        logging.info(f"Task '{task}' added successfully.")
+    except Exception as e:
+        logging.error("An error occurred while adding the task.")
+        logging.exception(e)
+
+# Feladat törlése a fájlból
+
+
+def remove_task(task: str):
+    tasks = read_tasks()
+    if task in tasks:
+        tasks.remove(task)
+        try:
+            with open(TASK_FILE, "w") as file:
+                for t in tasks:
+                    file.write(t + "\n")
+            logging.info(f"Task '{task}' removed successfully.")
+        except Exception as e:
+            logging.error("An error occurred while removing the task.")
+            logging.exception(e)
+    else:
+        logging.warning(f"Task '{task}' not found in the list.")
+
+
+def clear_all_task():
+    user_input = str(
+        input(f"Are you sure??? Press 'Y' to continue!: "))
```

Comment:
a bement érzékeny a kis és nagybetűkre, ezért érdemes normalizálni ezt:
`user_input = input("Are you sure? Press 'Y' to continue: ").strip().upper()`

Így ha felhasználó kis betűvel írja az y-t, vagy szóközt hagy a végén, akkor is lefut a megfelelő ág.

Reply:
Oksa, kiszedtem a .gitignore-t nem vettem észre hogy már van belőle egy a main-ban.
Valamit a environmet-ekkel kavartam (talán azt hogy nem "env" ként hoztam létre az environmentement, hanem valami "mylesson4env" vagy valami ilyesmi, és így olyan néven hozta létre a mappát, így meg a github desktop új file-oknak látta :)